java/memory/src/main/java/org/apache/arrow/memory/BaseAllocator.java

```diff
@@ -275,6 +275,10 @@ public ArrowBuf buffer(final int initialRequestSize, BufferManager manager) {
         nextPowerOfTwo(initialRequestSize)
         : initialRequestSize;
     AllocationOutcome outcome = this.allocateBytes(actualRequestSize);
+    if (!outcome.isOk() && listener.onFailedAllocation(actualRequestSize, outcome)) {
+      // Second try, in case the listener can do something about it
+      outcome = this.allocateBytes(actualRequestSize);
+    }
     if (!outcome.isOk()) {
```

Comment:
let's avoid checking twice in the base case.

Reply:
done